src/org/sosy_lab/java_smt/api/BasicProverEnvironment.java

```diff
@@ -22,9 +22,17 @@
  * provides only the common operations. In most cases, just use one of the two sub-interfaces
  */
 public interface BasicProverEnvironment<T> extends AutoCloseable {
-
   String NO_MODEL_HELP = "Model computation failed. Are the pushed formulae satisfiable?";
 
+  String NO_UNSAT_CORE_HELP =
+      "UnsatCore computation failed. Are the pushed formulae unsatisfiable?";
+
+  String STACK_CHANGED_HELP =
+      "Computation failed. The prover state has changed since the last call to isUnsat().";
+
+  // Used as prefix concatenated with the reason in the IllegalStateException thrown for shutdowns
+  String SHUTDOWN_EXCEPTION_PREFIX = "Prover is not usable due to interrupt with message: ";
```

Comment:
Are these 3 strings supposed to be part of your public API? If not, they should not be here. Every user of a prover environment will see them for auto-completion, for example. (Same for `NO_MODEL_HELP` already.)

Reply:
No, they are meant for us internally. This was just a central point to access them. I moved them to be hidden from users, but now we have `SHUTDOWN_EXCEPTION_PREFIX` double (but with info to keep it in sync). I could not find a way to share this field without breaking changes, may you know a way?

Reply:
I still see no explanation why adding a public constant somewhere (for example in `AbstractProver`) should be a breaking change.

But as mentioned, IMHO this shouldn't even be done. Please don't use strings and exception messages in order to convey information between classes. It is certainly a bad idea that one class relies on the specific format of the exception message produced by a different class. For example this breaks as soon as someone uses the `BasicProverWithAssumptionsWrapper` for an SMT solver that happens to not use the `AbstractProver` base class.

Reply:
The remaining problem has been removed, as the correct exception is thrown without workaround.